ibazel/profiler/profiler.js

```diff
@@ -0,0 +1,123 @@
+// Copyright 2017 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+(function(){
+  function setOption(options, name, value) {
+    if (typeof value === 'undefined') {
+      return;
+    }
+    if (!isNaN(+value)) {
+      value = +value;
+    }
+    return options[name] = value;
+  };
+
+  function extractOptions() {
+    var element, keyAndValue, m, mm, options, pair, src, _i, _j, _len, _len1, _ref, _ref1;
+    _ref = window.document.getElementsByTagName('script');
+    for (_i = 0, _len = _ref.length; _i < _len; _i++) {
+      element = _ref[_i];
+      if ((src = element.src) && (m = src.match(/^[^:]+:\/\/(.*)\/z?profiler\.js(?:\?(.*))?$/))) {
```

Comment:
Since you're in the context of the browser here, I would prefer if you used the browser's url parser to achieve this. I believe this is equivalent to the for loop contents.

```
const parsed = document.createElement('a');
parsed.href = element.src;
options.url = `${parsed.protocol}//${parsed.host}/profiler-event`;
// Note starts here
p.search.substring(1).split('&').forEach((e) => {
  [k, ...v] = e.split("=");
  options[k] = v.join("=");
})
```

Additionally after "// Note starts here" I'm not sure what the value of adding the query params is? Does that exist so that you can override options.{https,host,port}? If you don't care about that feel free to omit after `// Note starts here`

Reply:
Thanks.